File: app/Stikl.Web/Logging.fs
module Stikl.Web.Logging

open Serilog
open Serilog.Events
open Serilog.Formatting.Compact

let configure () : LoggerConfiguration =

    let logLevel =
        EnvironmentVariable.get "LOG_LEVEL"
        |> Option.defaultValue "INFORMATION"
        |> fun v -> LogEventLevel.Parse(v, ignoreCase = true)

    LoggerConfiguration()
        .MinimumLevel.Is(logLevel)
        .MinimumLevel.Override("Microsoft.AspNetCore.Hosting.Diagnostics", LogEventLevel.Warning)
        .MinimumLevel.Override("Microsoft.AspNetCore.Routing.EndpointMiddleware", LogEventLevel.Warning)
        .MinimumLevel.Override("Microsoft.AspNetCore.Http.Result.ContentResult", LogEventLevel.Warning)
        // TODO: use prettier variant locally.
        .WriteTo.Console(formatter = RenderedCompactJsonFormatter())

File: app/Stikl.Web/Logging.cs
using Serilog;
using Serilog.Events;
using Serilog.Formatting.Compact;
using Serilog.Sinks.SystemConsole.Themes;

public static class Logging
{
    public static LoggerConfiguration CreateConfiguration()
    {
        var loggerConfig = new LoggerConfiguration()
            .MinimumLevel.Is(EnvironmentVariable.LogLevel)
            .MinimumLevel.Override("Microsoft.AspNetCore", LogEventLevel.Warning)
            .MinimumLevel.Override("System.Net.Http.HttpClient", LogEventLevel.Warning)
            .MinimumLevel.Override("Microsoft.AspNetCore.Server.Kestrel", LogEventLevel.Warning)
            .MinimumLevel.Override("Microsoft.AspNetCore.Watch", LogEventLevel.Warning)
            .MinimumLevel.Override("System.Net.Http", LogEventLevel.Warning)
            .MinimumLevel.Override(
                "Microsoft.AspNetCore.StaticFiles.StaticFileMiddleware",
                LogEventLevel.Warning
            )
            .Enrich.FromLogContext()
            .WriteTo.Console(
                outputTemplate: "[{Timestamp} {Level:u3} {SourceContext}] {Message:lj}{NewLine} [{Properties:j}]{NewLine}{Exception}",
                theme: AnsiConsoleTheme.Code,
                applyThemeToRedirectedOutput: true
            )
            .WriteTo.Console(new RenderedCompactJsonFormatter());

        return loggerConfig;
    }
}
